packages/shared/client/ReplayClient.ts

```diff
@@ -332,6 +339,58 @@ export class ReplayClient implements ReplayClientInterface {
     client.Session.removeNavigationEventsListener(onNavigationEvents);
   }
 
+  async findPoints(
+    pointSelector: PointSelector,
+    pointLimits?: PointPageLimits
+  ): Promise<PointDescription[]> {
+    const points: PointDescription[] = [];
+    const sessionId = this.getSessionIdThrows();
+    const findPointsId = String(this.nextFindPointsId++);
+    pointLimits = pointLimits ? { ...pointLimits } : {};
+    if (!pointLimits.maxCount) {
+      pointLimits.maxCount = MAX_POINTS_TO_FIND;
+    }
+
+    await this._waitForRangeToBeLoaded(
+      pointLimits?.begin && pointLimits.end
+        ? { begin: pointLimits.begin, end: pointLimits.end }
+        : null
+    );
+
+    function onPoints(results: findPointsResults) {
+      if (results.findPointsId === findPointsId) {
+        points.push(...results.points);
+      }
+    }
+
+    addEventListener("Session.findPointsResults", onPoints);
+
+    let result;
+    try {
+      result = await client.Session.findPoints(
+        { pointSelector, pointLimits, findPointsId },
+        sessionId
+      );
```

Comment:
I'd recommend changing this to be
```
try {
  const maxCount = pointLimits?.maxCount ?? MAX_POINTS_TO_FIND;
  let begin = pointLimits?.begin ?? undefined;
  const end = pointLimits?.end ?? undefined;
  while (true) {
    const { pointPage } = await client.Session.findPoints(
      {
        findPointsId,
        pointSelector,
        pointLimits: {
          begin,
          end,
          // This should always be non-negative so but just adding for clarity.
          maxCount: Math.max(0, maxCount - points.length),
        },
      },
      sessionId
    );

    if (
      // Got to the end of the focus winow so definitely done.
      !pointPage.hasNext ||
      // We reached the end endpoint on the query so we're done.
      (end && compareNumericStrings(pointPage.end, end) >= 0) ||
    ) {
      break;
    }

    // If other parts of the codebase changed the focus window out from under 
    // this code this could happen, and if this isn't handled this code would be
    // at risk of being an infinite loop so it should be handled somehow.
    if (pointPage.begin === pointPage.end) {
      throw commandError("Too many points", ProtocolError.FocusWindowChange);
    }

    // If you hit the max count but didn't bail out for any of the cases
    // above then throw too many points.
    if (points.length >= maxCount) {
      throw commandError("Too many points", ProtocolError.TooManyPoints);
    }

    // Pick up querying after this page of results.
    begin = pointPage.end;
  }
} finally {
```
with the same type of thing for `runEvaluation`.

This has the great benefit that it separates the max results per page that the backend allows from the max count that the frontend wants to query for.